src/engraving/libmscore/score.cpp

```diff
@@ -2648,6 +2648,15 @@ void Score::insertStaff(Staff* staff, staff_idx_t ridx)
         }
     }
 
+    if (staff->isSystemObjectStaff()) {
+        bool isAlreadyInSystemObjectList = std::find_if(systemObjectStaves.begin(), systemObjectStaves.end(),
+                                                        [staff](Staff* stf){ return stf == staff; })
+                                           != systemObjectStaves.end();
```

Comment:
Couldn't you use `mu::contains(systemObjectStaves, staff)` here?

Reply:
Thanks! Didn't know we had them